mmaprototype: remove health tracking, do some housekeeping

tbg opened the pull request:
Epic: CRDB-55052

tbg committed:
mmaprototype: remove failureDetectionSummary

We'll replace it with a store-level construct in follow-up commits.

tbg committed:
mmaprototype: remove storeMembership

We'll replace it with a store-level construct in follow-up commits.

tbg committed:
mmaprototype: move and rename storeIDPostingList -> storeSet

tbg committed:
mmaprototype: fix a typo

tbg committed:
mmaprototype: move storeLoadSummary to its own file

tbg committed:
mmaprototype: add comment to storeLoadSummary

tbg committed:
mmaprototype: avoid shadowing

loadSummary is also a type. Linter complained in CI all out of
a sudden :shrug:

wenyihu6 reviewed (approved):
Thanks for taking care of the housekeeping tasks and the easy review as usual.  <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156933#-:-OdKMOS8E_Qs7fl1vhL-:bdmy7e3)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[`pkg/kv/kvserver/allocator/mmaprototype/allocator_state.go` line 816 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/156933#-OdKJjo8-C2_yhSFBa2M:-OdKJjo8-C2_yhSFBa2N:b-1m1lgk) ([raw file](https://github.com/cockroachdb/cockroach/blob/05c6218cc548c914bcf0925bde1d1a9000854056/pkg/kv/kvserver/allocator/mmaprototype/allocator_state.go#L816)):*
> ```Go
> 
> // RemoveNodeAndStores implements the Allocator interface.
> func (a *allocatorState) RemoveNodeAndStores(nodeID roachpb.NodeID) error {
> ```

Should we delete RemoveNodeAndStores as well? What is it for?
___
*[`pkg/kv/kvserver/allocator/mmaprototype/store_load_summary.go` line 13 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/156933#-OdKL7GvAkyxVetkeebf:-OdKL7GvAkyxVetkeebg:beuuzoo) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0ea6d4bf8e487a4aaebec07d77f2aa2b7615362/pkg/kv/kvserver/allocator/mmaprototype/store_load_summary.go#L13)):*
> ```Go
> // load across a set of permissible stores (often, all that satisfy the
> // constraints for a given range). Sources and targets are primarily picked
> // based on the store- and node level load summaries contained in this struct.
> ```

super nit: :s/-// (no need to push again for this - feel free to just squeeze this in future pr)


<!-- Sent from Reviewable.io -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> @sumeerbhola reviewed 6 of 10 files at r2, 1 of 2 files at r3, 6 of 10 files at r4, 2 of 2 files at r5, 2 of 3 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156933#-:-OdKnmvT2UmvX9fM0llh:bg7igkv)* status: <span>:shipit:</span> complete! 2 of 0 LGTMs obtained (waiting on @tbg)



<!-- Sent from Reviewable.io -->

tbg committed:
mmaprototype: remove RemoveNodeAndStores

It is unused and unclear what it would be used for.
Easy to re-add should a need emerge.

tbg committed:
mmaprototype: fix typo

tbg reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156933)* status: <span>:shipit:</span> complete! 2 of 0 LGTMs obtained (waiting on @wenyihu6)

___
*[`pkg/kv/kvserver/allocator/mmaprototype/allocator_state.go` line 816 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/156933#-OdKJjo8-C2_yhSFBa2M:-OdOC3bV0FbSNUelDeBW:b-f0p2qu) ([raw file](https://github.com/cockroachdb/cockroach/blob/05c6218cc548c914bcf0925bde1d1a9000854056/pkg/kv/kvserver/allocator/mmaprototype/allocator_state.go#L816)):*
<details><summary><i>Previously, wenyihu6 (Wenyi Hu) wrote…</i></summary><blockquote>

Should we delete RemoveNodeAndStores as well? What is it for?
</blockquote></details>

No idea. It is unused and unimplemented. Removed.


<!-- Sent from Reviewable.io -->
